mui/native_plugin.py

```diff
@@ -73,3 +77,107 @@ def __init__(self, start: int):
     def will_run_callback(self, ready_states):
         for state in ready_states:
             state.cpu.emulate_until(self.start)
+
+
+class TraceBlockPlugin(Plugin):
+    """Manticore plugin to record execution trace at a **block** level"""
+
+    def __init__(self, mui_state: MUIState):
+        super().__init__()
+        self.mui_state = mui_state
+
+    # Instructions that jump to new block
+    BLOCK_INS = {
```

Comment:
Right now I've manually selected the instructions that assign to cpu.PC in `x86.py`, but I'll look into generating this automatically

Reply:
I've looked into trying to generate this array in capstone, I think it could be done using by iterating through all the instruction ids for an arch [[ref]](https://github.com/capstone-engine/capstone/blob/441d37fbd60d755b12ecf0f7915f36c07c2761d1/include/capstone/x86.h#L384) and calling [`X86_get_insn_id`](https://github.com/capstone-engine/capstone/blob/63accbdc3f31d21f312cfa8b232e8ec9699c7486/arch/X86/X86Mapping.c#L2492). We can then inspect the results for the [`X86_GRP_JUMP`](https://github.com/capstone-engine/capstone/blob/63accbdc3f31d21f312cfa8b232e8ec9699c7486/arch/X86/X86Mapping.c#L2659) group being set in the instruction details.

Current blocker is I'm not sure how to do this through the python bindings for capstone. For now I will write more comments on the current manually generated list.